### PR TITLE
Mark results having submissions/answers and their ancestors as started

### LINKED ISSUE #627: Mark results having submissions/answers and their ancestors as started

### PR DESCRIPTION
**Pull request by zenovich**
Fixes #627

### CODE REVIEW

File: db/migrations/2011222206_mark_results_with_answers_and_hints_as_started.sql [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          `answers`.`item_id` = `results`.`item_id`),
+    `latest_submission_at`, `latest_hint_at`),
+    started_at = LEAST(`started_at`,
+        IFNULL(`latest_submission_at`, `latest_hint_at`),
+        IFNULL(`latest_hint_at`, `latest_submission_at`))
+WHERE (`latest_submission_at` IS NOT NULL OR `latest_hint_at` IS NOT NULL) AND `started_at` IS NULL;
+
+/*
+  Set started_at for results having descendant results with started_at set for the same attempt.
+  Items linked to both ancestor and descendant results must be visible for the participant.
```
Comment (smadbe) [inline db/migrations/2011222206_mark_results_with_answers_and_hints_as_started.sql:18]: I don't get this sentence. There are still visible even if not started. 🤔

**Review by smadbe — APPROVED** [review]

File: db/migrations/2011222206_mark_results_with_answers_and_hints_as_started.sql (hunk 1 above)
Comment (zenovich) [reply db/migrations/2011222206_mark_results_with_answers_and_hints_as_started.sql:18]: We do not start a parent result if the child result is started, but the parent (or child) item is not visible.

Comment (smadbe) [reply db/migrations/2011222206_mark_results_with_answers_and_hints_as_started.sql:18]: I don't get your answer.
But no worries, you can merge (we can discuss on slack)

Comment (zenovich) [reply db/migrations/2011222206_mark_results_with_answers_and_hints_as_started.sql]: fixed

**Pull request merged**